worlds/ahit/Regions.py

```diff
@@ -0,0 +1,946 @@
+    "Rock the Boat":               "Cruise_Sinking",
+    "Time Rift - Balcony":         "Cruise_WaterRift_Slide",
+    "Time Rift - Deep Sea":        "Cruise_CaveRift_Aquarium",
+
+    "Nyakuza Free Roam":            "MetroFreeRoam",  # not an actual Time Piece
+    "Rush Hour":                    "Metro_Escape",
+    "Time Rift - Rumbi Factory":    "Metro_CaveRift_RumbiFactory"
+}
+
+# Guarantee that the first level a player can access is a location dense area beatable with no items
+guaranteed_first_acts = [
+    "Welcome to Mafia Town",
+    "Barrel Battle",
+    "She Came from Outer Space",
+    "Down with the Mafia!",
+    "Heating Up Mafia Town",  # Removed in umbrella logic
+    "The Golden Vault",
+
+    "Contractual Obligations",  # Removed in painting logic
+    "Queen Vanessa's Manor",  # Removed in umbrella/painting logic
+]
+
+purple_time_rifts = [
+    "Time Rift - Mafia of Cooks",
+    "Time Rift - Dead Bird Studio",
+    "Time Rift - Sleepy Subcon",
+    "Time Rift - Alpine Skyline",
+    "Time Rift - Deep Sea",
+    "Time Rift - Tour",
+    "Time Rift - Rumbi Factory",
+]
+
+chapter_finales = [
+    "Dead Bird Studio Basement",
+    "Your Contract has Expired",
+    "The Illness has Spread",
+    "Rock the Boat",
+    "Rush Hour",
+]
+
+# Acts blacklisted in act shuffle
+# entrance: region
+blacklisted_acts = {
+    "Battle of the Birds - Finale A":   "Award Ceremony",
+}
+
+# Blacklisted act shuffle combinations to help prevent impossible layouts. Mostly for free roam acts.
+blacklisted_combos = {
+    "The Illness has Spread":           ["Nyakuza Free Roam", "Alpine Free Roam", "Contractual Obligations"],
+    "Rush Hour":                        ["Nyakuza Free Roam", "Alpine Free Roam", "Contractual Obligations"],
+
+    # Bon Voyage is here to prevent the cycle: Owl Express -> Bon Voyage -> Deep Sea -> MOTOE -> Owl Express
+    # which would make them all inaccessible since those rifts have no other entrances
+    "Time Rift - The Owl Express":      ["Alpine Free Roam", "Nyakuza Free Roam", "Bon Voyage!",
+                                         "Contractual Obligations"],
+
+    "Time Rift - The Moon":             ["Alpine Free Roam", "Nyakuza Free Roam", "Contractual Obligations"],
+    "Time Rift - Dead Bird Studio":     ["Alpine Free Roam", "Nyakuza Free Roam", "Contractual Obligations"],
+    "Time Rift - Curly Tail Trail":     ["Nyakuza Free Roam", "Contractual Obligations"],
+    "Time Rift - The Twilight Bell":    ["Nyakuza Free Roam", "Contractual Obligations"],
+    "Time Rift - Alpine Skyline":       ["Nyakuza Free Roam", "Contractual Obligations"],
+    "Time Rift - Rumbi Factory":        ["Alpine Free Roam", "Contractual Obligations"],
+
+    # See above comment
+    "Time Rift - Deep Sea":             ["Alpine Free Roam", "Nyakuza Free Roam", "Contractual Obligations",
+                                         "Murder on the Owl Express"],
+}
+
+
+def create_regions(world: "HatInTimeWorld"):
+    w = world
```

Comment:
i can't really condone going out of your way to make this harder to read

Reply:
This is something I did a long time ago to avoid going over the 120 char per line rule, but I suppose it's not really necessary here.